pulp_2to3_migration/app/pre_migration.py

```diff
@@ -77,12 +77,39 @@ def pre_migrate_content_type(content_model, mutable_type, lazy_type, premigrate_
         content_model: Models for content which is being migrated.
         mutable_type: Boolean that indicates whether the content type is mutable.
     """
+    def delete_removed_pulp2_content(content_model):
+        """
+        Delete Pulp2Content records for content which is no longer present in Pulp2.
+
+        This is to avoid situations and extra work when not all content migrated during the first
+        migration run, then orphan clean up is run in Pulp 2, and then migration is run again.
+
+        Args:
+            content_model: Pulp 2 content model
+
+        """
+        content_type = content_model.pulp2.TYPE_ID
+        mongo_content_qs = content_model.pulp2.objects().only('id')
+        mongo_content_ids = {c['_id'] for c in mongo_content_qs.as_pymongo().no_cache()}
+        premigrated_content_ids = set(
+            Pulp2Content.objects.filter(
+                pulp2_content_type_id=content_type
+            ).only('pulp2_id').values_list('pulp2_id', flat=True)
+        )
+        content_ids_to_delete = premigrated_content_ids - mongo_content_ids
+        Pulp2Content.objects.filter(
+            pulp2_content_type_id=content_type,
+            pulp2_id__in=content_ids_to_delete
+        ).delete()
```

Comment:
Instead, I could use `exclude`:
`Pulp2Content.objects.filter(pulp2_content_type_id=content_type).exclude(pulp2_id__in=mongo_content_ids).delete()`

But I'm not sure if it's better to pass a huge list of ids into exclude. Is there a limit on the query size?
Likely, in the majority of cases, `mongo_content_ids` list is big and the `content_ids_to_delete` is small.

Reply:
Using subqueries is almost always better than passing lists of IDs, if it can be helped.  Granted, this is Mongo, I'm less familiar with how it would behave.

Reply:
Mongo is only on line 92-93, and we get list of ids from mongo anyway, so we can't avoid passing it.
But for Pulp2Content (which is postgres), is it better to pass once a huge list, or make couple of queries and pass a smaller one later?

Reply:
Ah, right, I misread.  Honestly, I'm not sure, but I would maybe lean towards the way it's being done currently.  The extra query should be cheap-ish since it's only working with a single, indexed field.